jaxopt/_src/lbfgs.py

```diff
@@ -256,19 +262,34 @@ def init_state(self,
     Returns:
       state
     """
+    if isinstance(init_params, base.OptStep):
+      # this allows to maintain backward compatibility with the previous API
+      hessian_approx_kwargs = dict(
```

Comment:
`iter_num` is not Hessian-related so maybe just rename the variable to `kwargs`

Reply:
since I will initialize `stepsize` I will change it to `kwargs` not matter what.